WBooks/Models/Book.swift

```diff
@@ -8,10 +8,36 @@
 
 import Foundation
 
-struct Book {
+struct Book: Codable {
     
+    let id: Int
     let title: String
     let author: String
+    let genre: String
+    let year: String
     let image: String?
+    let status: String
     
+    public init(from: Decoder) throws {
+        let container = try from.container(keyedBy: BookKey.self)
+        id = try container.decode(Int.self, forKey: .id)
+        title = try container.decode(String.self, forKey: .title)
+        author = try container.decode(String.self, forKey: .author)
+        genre = try container.decode(String.self, forKey: .genre)
+        year = try container.decode(String.self, forKey: .year)
+        image = try container.decode(String.self, forKey: .image)
+        status = try container.decode(String.self, forKey: .status)
+    }
+
+}
+
+// Book Json keys enum
+enum BookKey: String, CodingKey {
+    case id
+    case title
+    case author
+    case genre
+    case year
+    case image
+    case status
```

Comment:
_Aclaración: tal vez te hayas dado cuenta que cuando le agregamos la extensión Codable a nuestro modelo no recibimos ningún error de que nuestra estructura no conforma al protocolo. Esto sucede por la misma razón que no necesitamos declarar un init() en los structs, y es que ya hay una función init(from: Decoder) default que se utiliza en caso de que nosotros no definamos la función. Esta función default utiliza como claves los nombres de los atributos de la estructura en string, por lo tanto si los nombres de los atributos de nuestra clase coinciden con las claves del JSON no necesitamos hacer nada de todo esto, solo agregar el protocolo a la estructura._

There is that comment in the tutorial. Please try it

Reply:
yeah yes, I had read it but I left it just in case 😆 
I'll remove it

Reply:
si quieres dejarlo para verlo despues, puedes cambiar el nombre de alguna de las propiedades, por ejemplo imageString, y así ya si tendrías que especificar el coding